docs/specs/m2m.md

```diff
@@ -0,0 +1,1206 @@
+
+In `https://inventory.myapp.com`, there is a daemon process that needs access to `https://onlinestore.myapp.com`.
+In particular, it needs to `read:orders`.
+
+`https://inventory.myapp.com` use `RFC6749 section-4.4` to obtain an `access_token` to gain access.
+In particular, it sends this Token request to `https://auth.myapp.com/oauth2/token`.
+
+```
+POST /oauth2/token HTTP/1.1
+Content-Type: application/x-www-form-urlencoded
+
+grant_type=client_credentials
+client_id=inventory
+client_secret=THE_CLIENT_SECRET
+resource=https://onlinestore.myapp.com
+scope=read:orders
+```
+
+`https://auth.myapp.com` process the Token request with these rules:
+
+- Check if `grant_type` is supported, and it is `client_credentials`.
+- Authenticate the client by checking `client_id` and `client_secret`.
+- Validate `resource` refers to an existing Resource, and the Resource is associated with `client_id`.
+- Validate `scope` and check if it refers to the valid values as defined in `resource`, and is associated with `client_id`.
+
+`https://auth.myapp.com` will return an `access_token` like this:
+
+```json
+{
+  "iss": "https://auth.myapp.com",
+  "sub": "client_id_inventory",
+  "aud": ["https://onlinestore.myapp.com"],
+  "client_id": "inventory",
+  "scope": "read:orders",
+  "https://authgear.com/claims/scope_by_aud": [
+    {
+      "aud": "http://onlinestore.myapp.com",
+      "scope": "read:orders"
+    }
+  ]
+}
+```
+
+If you compare this `access_token` to the `access_token` of the end-user, you will notice:
+
+|             | Description                      | `access_token` of end-user                                     | `access_token` of client                                                             |
+| ---         | ---                              | ---                                                            | ---                                                                                  |
+| `iss`       | Same value                       |                                                                |                                                                                      |
+| `sub`       | Different value                  | The user ID (`johndoe`)                                        | The string concatenation of `client_id_` and the `client_id` (`client_id_inventory`) |
+| `aud`       | The target audience of the token | The authentication server plus `resource`                      | The `resource` parameter (`https://onlinestore.myapp.com`)                           |
+| `client_id` | Different value                  | The client acting on behalf of an end-user (`mobileapp`)       | The client acting on behalf of itself (`inventory`)                                  |
+| `scope`     | Same meaning                     | The access of the `client_id` to `aud` on `sub`. See Remarks 1 | The access of the `client_id` to `aud` on behalf of `sub`. See Remarks 2             |
```

Comment:
I just found that our user access token does not have scope

Reply:
https://datatracker.ietf.org/doc/html/rfc9068#section-2.2.3

It is not required to have it. In this specification, I changed the behavior to always include it.